### ISSUE TITLE
Operation .timeout - possibly not working?

### ISSUE THREAD

**Issue by iandundas** [ReactiveKit/ReactiveKit#73]
Hey - using v2.0.1:

I'd expect the following to send an error event to the `.observe` block after five seconds, but actually the observe block is never called with any event. I've used an empty Operation to demonstrate that nothing will be sent. The `bag` is definitely still retained. The documentation states: `Error-out if interval time passes with no emitted elements.`. Is this a bug or have I misunderstood? :) thanks 

```swift
private let bag = DisposeBag()

func timeoutTest(){
    let timeoutOperation:Operation<Void, NSError> = Operation { observer in
        // do nothing
        return SimpleDisposable()
    }
    
    let error = NSError(domain: "", code: 0, userInfo: [:])
    
    timeoutOperation
        .timeout(5, with: error, on: Queue.main)
        .observe { event in
            print("did timeout..")
        }.disposeIn(bag)
    
}
```

**Comment by srdanrasic** [comment]
Hi @iandundas, thank you for reporting that. It looks like a bug. I'll try to push a fix today.

**Comment by iandundas** [comment]
awesome, thanks :)

**Comment by srdanrasic** [comment]
You're welcome! Should be fixed now (v2.0.2).

**Comment by iandundas** [comment]
Great, that works now 👍🏻thanks

**Comment by iandundas** [comment]
hey again,

```swift
private let bag = DisposeBag()

func timeoutTest(){
    let timeoutOperation:Operation<Void, NSError> = Operation { observer in
        // do nothing much
        
        return BlockDisposable {
            // Release any resources here (i.e. after timeout fails)
            print("did dispose")
        }
    }
    
    let error = NSError(domain: "", code: 0, userInfo: [:])
    
    timeoutOperation
        .timeout(2, with: error, on: Queue.main)
        .observe { event in
            // we probably timed out, because the Operation never sends anything in this example.. 
        }.disposeIn(bag)
}
```

Let's say the `timeoutOperation` owns some resources - not shown (in my case, `CLLocationManager` is locating). If this operation fails due to a timeout (`.timeout(2, with: error, on: Queue.main)`), we correctly receive a `.Failure` event in our observe block, but as the `bag: DisposeBag` is not deallocated at this time, the timeoutOperation observation is never disposed, so the resources are not released (i.e. for me, CLLocationManager keeps going and going..).

Basically, `BlockDisposable` block is never called after timeout.

My expectation was that the timeout would cause the Operation to be disposed & thus free any resources -  but I realise this is prevented by the `DisposeBag()` not being deallocated.

What would be the best way to use the timeout to stop/release resources that the operation was using?

Thanks!

**Comment by srdanrasic** [comment]
This is something I was thinking about, but needed a confirmation which you just gave me :)

Streams and operations are not disposed on completion or failure. Check out v2.0.3.

**Comment by iandundas** [comment]
Perfect - works now. Nice one! 🙏
